framework/wazuh/core/cluster/master.py

```diff
@@ -58,7 +58,8 @@ def done_callback(self, future=None):
             Synchronization process result.
         """
         super().done_callback(future)
-        # Integrity task is only freed if master is not waiting for Extra valid files.
+        # Integrity task is only freed if master is not waiting for Extra valid files. This condition should always take
+        # place
         if not self.wazuh_common.extra_valid_requested:
             self.wazuh_common.sync_integrity_free[0] = True
```

Comment:
If we could isolate the extra-valid functionality we could reduce the complexity of the code and this type of checks, thus eliminating some overhead in each iteration of integrity.